src/biaplotter/artists.py

```diff
@@ -236,26 +247,104 @@ def color_indices(self, indices: np.ndarray):
+                        'Categorical colormap detected. Setting color normalization method to linear.')
+                    self._color_normalization_method = 'linear'
+                norm = Normalize(vmin=0, vmax=self.overlay_colormap.N)
+            else:
+                with warnings.catch_warnings():
+                    # If array is all NaN, ignore warning since it will be displayed transparent
+                    warnings.filterwarnings(
+                        'ignore', r'All-NaN slice encountered')
+                    if self._color_normalization_method == 'log':
+                        min_value = np.nanmin(self._color_indices)
+                        if min_value <= 0:
+                            min_value = 0.01
+                        norm = norm_class(vmin=min_value, vmax=np.nanmax(self._color_indices))
+                        warnings.warn(
+                            f'Log normalization applied to color indices with min value {min_value}. Values below 0.01 were set to 0.01.')
+                        indices[indices <= 0] = min_value
+                    elif self._color_normalization_method == 'centered':
+                        norm = norm_class(vcenter=np.nanmean(self._color_indices))
+                    elif self._color_normalization_method == 'symlog':
+                        norm = norm_class(vmin=np.nanmin(self._color_indices), vmax=np.nanmax(self._color_indices), linthresh=0.03)
+                    else:
+                        norm = norm_class(vmin=np.nanmin(self._color_indices), vmax=np.nanmax(self._color_indices))
+
+            # Create a ScalarMappable instance using chosen colormap and normalization
+            sm = ScalarMappable(norm=norm, cmap=self.overlay_colormap.cmap)
+            # Convert normalized data to RGBA
+            rgba_colors = sm.to_rgba(indices)
+            # Set color light gray if overlay is not visible
+            if not self._overlay_visible:
+                rgba_colors = cat10_mod_cmap(0) # Set colors to light gray
+            self._scatter.set_facecolor(rgba_colors)
             self._scatter.set_edgecolor(None)
         # emit signal
         self.color_indices_changed_signal.emit(self._color_indices)
         self.draw()
 
     @property
-    def alpha(self) -> Union[float, np.ndarray]:
-        """Gets or sets the alpha value of the scatter plot.
+    def overlay_colormap(self) -> BiaColormap:
+        """Gets or sets the overlay colormap for the scatter plot.
 
-        Triggers a draw idle command.
+        Returns
+        -------
+        overlay_colormap : BiaColormap
+            colormap for the scatter plot with a `categorical` attribute.
+        """
+        return self._overlay_colormap
+    
+    @overlay_colormap.setter
+    def overlay_colormap(self, value: Colormap):
+        """Sets the overlay colormap for the scatter plot."""
+        self._overlay_colormap = BiaColormap(value)
+        self.color_indices = self._color_indices
```

Comment:
I think this ends up being problematic for me in some points. Why does changing the colormap require setting the `color_indices`? In the scope of the napari clusters plotter, this triggers an infinite loop.

Cascade:
1. color_indeces are changed (e.g., when a hue value is selected)
2. `color_indeces_changed` is emitted
3. layer recoloring is triggered
4. layer coloring inspects the hue value to color by and changes the colormap of the plot accordingly, if necessary
5. `color_indeces` is overwritten here
6. `color_indeces_changed` is emitted --> infinte loop